Ofono template: Add missing properties to SimManager interface

pete-woods opened the pull request.

martinpitt commented on dbusmock/templates/ofono.py:
```diff
@@ -47,6 +49,19 @@ def load(mock, parameters):
         mock.AddModem(parameters.get('ModemName', 'ril_0'), {})
 
 
+def newSimSerial():
```
`python_style`, not `camelCase` please. This isn't a D-Bus method.

martinpitt commented on dbusmock/templates/ofono.py:
```diff
@@ -47,6 +49,19 @@ def load(mock, parameters):
 
+def newSimSerial():
+    global _simSerialCounter
+    serial = 'ed752c5f-f723-437e-bc6c-' + ('%012d' % _simSerialCounter)
```
This this serial magic in any way? I. e. does ofono do something special for this prefix? If so, please add a comment. If it's just a random number, maybe 12345..abcd? (but that's just a nitpick)

martinpitt commented on dbusmock/templates/ofono.py:
```diff
@@ -47,6 +49,19 @@ def load(mock, parameters):
+
+def newImsi():
+    global _imsiCounter
+    imsi = '310150' + ('%09d' % _imsiCounter)
```
same two issues/questions.

martinpitt commented on dbusmock/templates/ofono.py:
```diff
@@ -79,6 +94,7 @@ def AddModem(self, name, properties):
                        'Manufacturer': dbus.String('Fakesys', variant_level=1),
                        'Model': dbus.String('Mock Modem', variant_level=1),
                        'Revision': dbus.String('0815.42', variant_level=1),
+                       'Serial': dbus.String(newSimSerial(), variant_level=1),
```
So `newSimSerial()` is only being called once, ever. Why the overhead with the function and counter instead of just setting it to a static value and letting the test case drive/set it? Mocks shouldn't be doing unnecessary things by themselves, especially not if the "real thing" doesn't do the same. And I doubt that the serial number changes?

martinpitt commented on dbusmock/templates/ofono.py:
```diff
@@ -304,12 +320,18 @@ def add_simmanager_api(mock):
         'PinRequired': _parameters.get('PinRequired', "none"),
         'SubscriberNumbers': _parameters.get('SubscriberNumbers', ['123456789', '234567890']),
-        'SubscriberIdentity': _parameters.get('SubscriberIdentity', "23456"),
+        'SubscriberIdentity': _parameters.get('SubscriberIdentity', newImsi()),
```
Same question here as for `Serial`.

martinpitt replied:
If you do need some dynamic stuff here, please add a simple test case.

pete-woods commented:
Okay, I've tried to address all your comments.

- I used snake_casing throughout now.
- Added comments to make it clear why we are generating the IMSI / modem serial.
- Made the counters be member variables so they get reset for each test case.
- Added a basic that shows the counter increments as we add new modems + SIMs.

martinpitt commented on dbusmock/templates/ofono.py:
```diff
@@ -102,13 +105,30 @@ def AddModem(self, name, properties):
+# Use a counter so that the result is predictable for tests.
+def new_modem_serial(mock):
+    serial = '12345678-1234-1234-1234-' + ('%012d' % mock.modem_serial_counter)
+    mock.modem_serial_counter = mock.modem_serial_counter + 1
```
`+= 1` please, to avoid the repetition of the long var, and easier to read.

martinpitt commented on dbusmock/templates/ofono.py:
```diff
@@ -102,13 +105,30 @@ def AddModem(self, name, properties):
+# Use a counter so that the result is predictable for tests.
+def new_imsi(mock):
+    imsi = '310150' + ('%09d' % mock.imsi_counter)
+    mock.imsi_counter = mock.imsi_counter + 1
```
dito

martinpitt commented on dbusmock/__init__.py:
```diff
@@ -11,7 +11,7 @@
 __copyright__ = '(c) 2012 Canonical Ltd.'
 __license__ = 'LGPL 3+'
-__version__ = '0.16.3'
+__version__ = '0.16.4'
```
This gets done automatically on release, commits shouldn't touch this. Please revert.

martinpitt commented:
Very nice, thanks! Only three nitpicks, but I'm happy to do them myself during merging, unless you beat me to them. Will merge/test/release tomorrow morning (sorry, running late today).

martinpitt commented:
Merged with the above nitpick changes. Thanks!
